apps/server/src/modules/room/api/test/room-get-room-invitation-link-list.api.spec.ts

```diff
@@ -0,0 +1,173 @@
+	});
+
+	describe('GET /:room-id/room-invitation-links', () => {
+		describe('when the user is not authenticated', () => {
+			it('should return a 401 error', async () => {
+				const someId = new ObjectId().toHexString();
+				const response = await testApiClient.get(someId);
+				expect(response.status).toBe(HttpStatus.UNAUTHORIZED);
+			});
+		});
+
+		describe('when the feature is disabled', () => {
+			const setup = async () => {
+				config.FEATURE_ROOM_INVITATION_LINKS_ENABLED = false;
+
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return a 403 error', async () => {
+				const { loggedInClient } = await setup();
+				const someId = new ObjectId().toHexString();
+				const response = await loggedInClient.get(`${someId}/room-invitation-links`);
+				expect(response.status).toBe(HttpStatus.FORBIDDEN);
+			});
+		});
+
+		describe('when id is not a valid mongo id', () => {
+			const setup = async () => {
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+				await em.persistAndFlush([studentAccount, studentUser]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient };
+			};
+
+			it('should return a 400 error', async () => {
+				const { loggedInClient } = await setup();
+				const response = await loggedInClient.get('42/boards');
+				expect(response.status).toBe(HttpStatus.BAD_REQUEST);
+			});
+		});
+
+		describe('when the user has the required permissions', () => {
+			const setup = async () => {
+				const school = schoolEntityFactory.buildWithId();
+				const room = roomEntityFactory.build({ schoolId: school.id });
+				const roomInvitationLinks = roomInvitationLinkEntityFactory.buildList(3, {
+					roomId: room.id,
+				});
+				const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent({ school });
+				const { roomAdminRole } = RoomRolesTestFactory.createRoomRoles();
+				const userGroupEntity = groupEntityFactory.buildWithId({
+					type: GroupEntityTypes.ROOM,
+					users: [{ role: roomAdminRole, user: studentUser }],
+					organization: studentUser.school,
+					externalSource: undefined,
+				});
+				const roomMembership = roomMembershipEntityFactory.build({
+					userGroupId: userGroupEntity.id,
+					roomId: room.id,
+					schoolId: school.id,
+				});
+				await em.persistAndFlush([
+					room,
+					...roomInvitationLinks,
+					studentAccount,
+					studentUser,
+					roomAdminRole,
+					userGroupEntity,
+					roomMembership,
+				]);
+				em.clear();
+
+				const loggedInClient = await testApiClient.login(studentAccount);
+
+				return { loggedInClient, room, roomInvitationLinks };
+			};
+
+			describe('when the room exists and has links', () => {
```

Comment:
for these "inner" describes I would recommend some structure like `and the room exists and has links`. Which also improves readability.

Reply:
the indentation in the code and in the test summary makes it easy to understand. I would not use "and" here, but I am open for a discussion.